src/Compilers/CSharp/Test/Emit3/Semantics/ExtensionTests.cs

```diff
@@ -21061,6 +21058,917 @@ static class Extensions
+            //     public static implicit operator int(long x) => throw null;
+            Diagnostic(ErrorCode.ERR_OperatorInStaticClass, "int").WithArguments("Extensions.implicit operator int(long)").WithLocation(3, 37),
+            // (7,30): error CS0111: Type 'Extensions' already defines a member called 'op_Implicit' with the same parameter types
+            //         static public string op_Implicit(long x) => throw null;
+            Diagnostic(ErrorCode.ERR_MemberAlreadyExists, "op_Implicit").WithArguments("op_Implicit", "Extensions").WithLocation(7, 30)
+            );
+    }
+
+    [Fact]
+    public void SignatureConflict_39_DifferentReturn_ConflictWithConversion()
+    {
+        var src1 = """
+public static class Extensions
+{
+    extension(int receiver)
+    {
+        static public string op_Implicit(long x) => throw null;
+    }
+
+    public static implicit operator int(long x) => throw null;
+}
+""";
+        var comp = CreateCompilation(src1);
+        comp.VerifyDiagnostics(
+            // (5,30): error CS0111: Type 'Extensions' already defines a member called 'op_Implicit' with the same parameter types
+            //         static public string op_Implicit(long x) => throw null;
+            Diagnostic(ErrorCode.ERR_MemberAlreadyExists, "op_Implicit").WithArguments("op_Implicit", "Extensions").WithLocation(5, 30),
+            // (8,37): error CS0715: 'Extensions.implicit operator int(long)': static classes cannot contain user-defined operators
+            //     public static implicit operator int(long x) => throw null;
+            Diagnostic(ErrorCode.ERR_OperatorInStaticClass, "int").WithArguments("Extensions.implicit operator int(long)").WithLocation(8, 37)
+            );
+    }
+
+    [Fact]
+    public void SignatureConflict_40_StaticVsInstance()
+    {
+        var src1 = """
+public static class Extensions
+{
+    extension(int receiver)
+    {
+        static public string M1(long x) => throw null;
+        public string M2() => throw null;
+    }
+
+    extension(long receiver)
+    {
+        public string M1() => throw null;
+        static public string M2(int x) => throw null;
+    }
+}
+""";
+        var comp = CreateCompilation(src1);
+        comp.VerifyDiagnostics(
+            // (11,23): error CS0111: Type 'Extensions' already defines a member called 'M1' with the same parameter types
+            //         public string M1() => throw null;
+            Diagnostic(ErrorCode.ERR_MemberAlreadyExists, "M1").WithArguments("M1", "Extensions").WithLocation(11, 23),
+            // (12,30): error CS0111: Type 'Extensions' already defines a member called 'M2' with the same parameter types
+            //         static public string M2(int x) => throw null;
+            Diagnostic(ErrorCode.ERR_MemberAlreadyExists, "M2").WithArguments("M2", "Extensions").WithLocation(12, 30)
+            );
+    }
```

Comment:
Do we expect this error?
```
    [Fact]
    public void TODO2()
    {
        var src = """
static class E
{
    extension(object)
    {
        public void Method() {  }
        public static void Method() { }
    }
}
""";
        var comp = CreateCompilation(src);
        comp.VerifyEmitDiagnostics(
            // (6,28): error CS0111: Type 'E.extension(object)' already defines a member called 'Method' with the same parameter types
            //         public static void Method() { }
            Diagnostic(ErrorCode.ERR_MemberAlreadyExists, "Method").WithArguments("Method", "E.extension(object)").WithLocation(6, 28));
    }
```

Reply:
> Do we expect this error?

Given the current rules, we do. This looks like `M9` scenario in ```SignatureConflict_02```. There is a PROTOTYPE comment regarding the current behavior

Reply:
The `M9` scenario in `SignatureConflict_02` is:
```
    extension(int receiver)
    {
        public void M9() {}
    }

    extension(int receiver)
    {
        public static void M9(int x) {}
    }
```

`M9` generates conflicting signatures in metadata, so should generate an error. 
But in the scenario above, the implementation methods have different signatures, so I would expect no error.

Reply:
> so I would expect no error.

Are both methods static extensions?